### PR TITLE
Possibility to pass instance profile name

### PR DESCRIPTION
**Pull request by ramirantala**
This option will allow passing existing instance profile name (=iam role) when launching instances. 

That would make possible to configure backups to s3 for example.

The instance profile name is optional.

### CODE REVIEW

File: create_cluster.py [hunk 1]
```diff
@@ -566,14 +572,15 @@ def print_failure_message():
 @click.option('--environment', '-e', multiple=True)
 @click.option('--sns-topic', help='SNS topic name to send Auto-Recovery notifications to')
 @click.option('--sns-email', help='Email address to subscribe to Auto-Recovery SNS topic')
+@click.option('--instance-profile-name', help='Instance profile name to be used, optional')
```
Comment (a1exsh) [inline create_cluster.py]: Other options don't mention 'optional'.  Could we update README and mention it there somehow?

**Comment by a1exsh** [comment]
Thank you for the PR! :-)

File: create_cluster.py (hunk 1 above)
Comment (ramirantala) [inline create_cluster.py]: Yes, sure..

Comment (ramirantala) [inline create_cluster.py]: Added documentation and removed optional from options help

**Comment by a1exsh** [comment]
:+1:

**Comment by a1exsh** [comment]
:+1:

**Comment by LappleApple** [comment]
👍

**Comment by LappleApple** [comment]
👍

**Comment by LappleApple** [comment]
@a1exsh @hjacobs This PR needed a Zappr approval; I approved but now it has branch conflicts. What is the plan?

**Comment by a1exsh** [comment]
This now unfortunately conflicts with the master branch, because instance profile is created by the script itself.  An option to pass existing instance profile will be still useful, we plan to add this at some point.

File: create_cluster.py [hunk 2]
```diff
@@ -426,6 +431,7 @@ def launch_instance(region: str, ip: dict, ami: object, subnet_id: str,
             SubnetId=subnet_id,
             PrivateIpAddress=ip['PrivateIp'],
             BlockDeviceMappings=block_devices,
+            IamInstanceProfile=instance_profile_name,
```
Comment (a1exsh) [inline create_cluster.py]: This is not going to work.  You need a dict with `Arn` or `Name`, see: http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.Client.run_instances

Comment (a1exsh) [inline create_cluster.py]: Ah, forget it, I've overlooked the assignment above.  Then I must say the variable name is quite confusing. ;)

**Pull request closed without merging**